compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -3734,6 +3735,12 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     // We skip some common traits that we don't want to consider because autoderefs
                     // would take care of them.
                     && !skippable.contains(&Some(pick.item.container_id(self.tcx)))
+                    && !skippable.contains(&pick.item.impl_container(self.tcx).and_then(|did| {
+                        match self.tcx.type_of(did).instantiate_identity().kind() {
+                            ty::Adt(def, _) => Some(def.did()),
+                            _ => None,
+                        }
+                    }))
```

Comment:
we've already manually filtered `as_ref` here. We now also check whether the impl self type is `Pin`. I don't really know what behavior I would expect here.

I think using `skippable` here is not great as it was previously only used for traits. Given that we're explicitly in code for `Pin`, it feels simpler to literally check ` whether the self type of the impl is `Pin`.

Reply:
changed